sous-chef/sous_chef/formatter/ingredient/get_ingredient_field.py

```diff
@@ -72,14 +108,46 @@ def parse_ingredient_field(self, ingredient_field):
         )
 
     @BaseWithExceptionHandling.ExceptionHandler.handle_exception
-    def _format_referenced_recipe(self, line: str):
-        referenced_recipe = self.ingredient_formatter.format_referenced_recipe(
+    def _format_referenced_recipe(self, source_recipe_title: str, line: str):
+        needed_ref_recipe = self.ingredient_formatter.format_referenced_recipe(
             line
         )
-        recipe = self.recipe_book.get_recipe_by_title(referenced_recipe.title)
-        recipe.factor *= referenced_recipe.quantity
-        recipe.amount = referenced_recipe.amount
-        self.referenced_recipe_list.append(recipe)
+        ref_recipe = self.recipe_book.get_recipe_by_title(
+            needed_ref_recipe.title
+        )
+
+        if needed_ref_recipe.pint_unit != unit_registry.dimensionless:
+            if (
+                ref_recipe.quantity.units.dimensionality
+                == needed_ref_recipe.pint_unit.dimensionality
+            ):
+                (
+                    needed_ref_quantity,
+                    needed_ref_units,
+                ) = UnitFormatter.convert_to_desired_unit(
+                    needed_ref_recipe.quantity,
+                    needed_ref_recipe.pint_unit,
+                    ref_recipe.quantity.units,
+                )
+                factor_dimensionless = (
+                    needed_ref_quantity * needed_ref_units / ref_recipe.quantity
+                )
+
+                ref_recipe.factor = factor_dimensionless.magnitude
```

Comment:
should we do *=?

Reply:
yes, seems more correct. factor is initialized to 1, which needs to remain.